File: backend/app/models/azure.py
from pydantic import BaseModel


class AzureTenant(BaseModel):
    tenantId: str
    displayName: str | None = None


class AzureSubscription(BaseModel):
    subscriptionId: str
    displayName: str
    state: str | None = None


class DatabricksWorkspace(BaseModel):
    id: str
    name: str
    location: str
    workspaceUrl: str | None = None
    properties: dict | None = None

File: frontend/src/types/azure.ts
export interface AzureTenant {
  tenantId: string
  displayName?: string
}

export interface AzureSubscription {
  subscriptionId: string
  displayName: string
  state?: string
}

export interface DatabricksWorkspace {
  id: string
  name: string
  location: string
  workspaceUrl?: string
  properties?: Record<string, unknown>
}
